benchmark/kernel/main_performance.py

```diff
@@ -37,40 +47,90 @@
 
 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
 
-# Decorate train and eval functions:
-train = profileit(print_layer_stats=False)(train)
-eval_acc = timeit()(eval_acc)
 
-for dataset_name, Net in product(datasets, nets):
-    dataset = get_dataset(dataset_name, sparse=True)
-    num_train = int(len(dataset) * 0.8)
-    num_val = int(len(dataset) * 0.1)
-
-    train_dataset = dataset[:num_train]
-    val_dataset = dataset[num_train:num_train + num_val]
-    test_dataset = dataset[num_train + num_val:]
-
-    train_loader = DataLoader(train_dataset, batch_size=args.batch_size,
-                              shuffle=True)
-    val_loader = DataLoader(val_dataset, batch_size=args.batch_size,
-                            shuffle=False)
-    test_loader = DataLoader(test_dataset, batch_size=args.batch_size,
-                             shuffle=False)
-
-    for num_layers, hidden in product(layers, hiddens):
-        print(f'--\n{dataset_name} - {Net.__name__} - {num_layers} - {hidden}')
-
-        model = Net(dataset, num_layers, hidden).to(device)
-        optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
-
-        stats_list = []
-        for epoch in range(1, args.epochs + 1):
-            loss, stats = train(model, optimizer, train_loader)
-            val_acc, val_time = eval_acc(model, val_loader)
-            test_acc, test_time = eval_acc(model, test_loader)
-
-            if epoch >= args.warmup_profile:
-                stats_list.append(stats)
-
-        stats_summary = get_stats_summary(stats_list)
-        print(stats_summary)
+def run_train():
```

Comment:
Same here. Is it possbile to merge the `run_train` and `run_inference` into one function?

Reply:
Hi, I think `run_train` and `run_inference` can be seperate.
1. Merging `run_train` and `run_inference` will make the function too large, and need `if` to distinguish `train` and `inference`. 
2. Most of the code in `run_inference` is not same with `run_train`.
3. Matthias asked to split into two functions in https://github.com/pyg-team/pytorch_geometric/pull/4892#discussion_r916699601 in benchmark/citation.

What do you think about this?

Reply:
I prefer to move the dataset and data loader part out of the `run_train` and `run_inference` functions but no objection to this.

Reply:
Done. I move the dataset and dataloader part to `prepare_dataloader`, and call in `run_train` and `run_inference`.